global_segment_map_node/src/controller.cpp

```diff
@@ -1248,45 +1233,49 @@ void Controller::publishScene() {
 // TODO(ntonci): Generalize this and rename since this one is also publishing
 // and saving, not just generating.
 void Controller::generateMesh(bool clear_mesh) {  // NOLINT
-  voxblox::timing::Timer generate_mesh_timer("mesh/generate");
   {
-    std::lock_guard<std::mutex> updateMeshLock(updated_mesh_mutex_);
-    if (clear_mesh) {
-      constexpr bool only_mesh_updated_blocks = false;
-      constexpr bool clear_updated_flag = true;
-      mesh_label_integrator_->generateMesh(only_mesh_updated_blocks,
-                                           clear_updated_flag);
-      if (enable_semantic_instance_segmentation_) {
-        all_semantic_labels_.clear();
-        mesh_semantic_integrator_->generateMesh(only_mesh_updated_blocks,
+    std::lock_guard<std::mutex> mesh_layer_lock(mesh_layer_mutex_);
+    voxblox::timing::Timer generate_mesh_timer("mesh/generate");
+    {
+      std::lock_guard<std::mutex> label_tsdf_layers_lock(
+          label_tsdf_layers_mutex_);
```

Comment:
Do you need two locks here `mesh_layer_lock` and `label_tsdf_layers_lock`?

Reply:
Yes, because the function MeshLabelIntegrator::generateMesh both
- Reads the TSDF and Label layers to mesh them
- Writes the Mesh layer.